### PR TITLE
[fast-ut] [reduced-it] Add BloomFilterMightContain corner cases [databricks]

### LINKED ISSUE #15618: [AutoSparkUT] Add BloomFilterMightContain corner-case IT coverage
## Coverage gap

The existing runtime bloom-filter join test verifies GPU replacement, but its probe side only contains `spark.range` values. It did not explicitly exercise nullable LONG corner values through `GpuBloomFilterMightContain`.

## Test change

Extend the existing `test_bloom_filter_join` parameterization with a literal nullable `LongGen` for `LONG_MIN`, `LONG_MAX`, zero, one, negative one, and null. Build exactly those six rows with an explicit Long schema and union them onto the large probe side, so the test remains focused and Spark runtime bloom-filter rewriting pushes `might_contain` onto the corner-data branch.

## Local validation evidence

Spark 3.3.0 / Python 3.10, current head `e952913b4`:

```text
16 passed, 34861 deselected, 6 warnings in 40.19s
```

The 16-test `-k test_bloom_filter_join` run includes all four modified combinations. Their capture assertions find both `GpuBloomFilterMightContain` and `GpuBloomFilterAggregate`.

Spark 3.5.0 / shim350, exact four modified combinations:

```text
4 passed, 34879 deselected, 6 warnings in 23.72s
covered lines: +0
covered branches: +1
```

### PR DESCRIPTION
**Pull request by wjxiz1992**
**JaCoCo sql-plugin line coverage: +0 lines** (shim350 vs nightly b14; +1 branch)

Closes #15618

## Summary

- Extend the existing runtime Bloom filter join IT with literal nullable Long probe data.
- Cover exactly `LONG_MIN`, `LONG_MAX`, zero, one, negative one, and null through an explicit Long schema rather than generating 2,048 random rows.
- Preserve the existing single-column/multi-column and batch-size matrix.
- Require both `GpuBloomFilterMightContain` and `GpuBloomFilterAggregate` in every modified GPU plan.

## Traceability

- RAPIDS test: `integration_tests/src/main/python/join_test.py::test_bloom_filter_join`
- Existing coverage extended in place; no duplicate test function was added.
- The executed GPU plan applies `might_contain(...)` to the corner-data branch.

## Validation

Spark 3.3.0 / Python 3.10, current head `e952913b4`:

```text
16 passed, 34861 deselected, 6 warnings in 40.19s
```

Spark 3.5.0 / shim350, exact four modified combinations, current Python test code:

```text
4 passed, 34879 deselected, 6 warnings in 23.72s
covered lines: +0
covered branches: +1
```

Documentation
- [ ] Updated for new or modified user-facing features or behaviors
- [x] No user-facing change

Testing
- [x] Added or modified tests to cover new code paths
- [ ] Covered by existing tests
      (Please provide the names of the existing tests in the PR description.)
- [ ] Not required

Performance
- [ ] Tests ran and results are added in the PR description
- [ ] Issue filed with a link in the PR description
- [x] Not required

### CODE REVIEW

File: integration_tests/src/main/python/join_test.py [hunk 1]
```diff
@@ -1363,14 +1363,17 @@ def test_hash_join_different_key_integral_types_asymmetric(left_gen, right_gen,
     else ()
 )
 
-def check_bloom_filter_join(confs, expected_classes, is_multi_column):
+def check_bloom_filter_join(confs, expected_classes, is_multi_column, probe_gen=None):
     def do_join(spark):
+        left = spark.range(100000)
+        if probe_gen is not None:
+            corner_rows = unary_op_df(spark, probe_gen).select(col("a").alias("id"))
+            left = left.unionByName(corner_rows)
```
[the remark this answers is not in the document]
Comment (wjxiz1992) [reply integration_tests/src/main/python/join_test.py:1377]: Thanks — updated in e952913b4. The probe side now unions exactly LONG_MIN, LONG_MAX, 0, 1, -1, and null through an explicit Long schema. Spark 3.3 / Python 3.10 validation: 16 passed, 0 failed.

**Review by thirtiseven — APPROVED** [review]
LGTM

**Comment by wjxiz1992** [comment]
build

**Comment by wjxiz1992** [comment]
build
